pontoon/base/utils.py

```diff
@@ -657,3 +657,91 @@ def build_translation_memory_file(creation_date, locale_code, entries):
         u'\n\t</body>'
         u'\n</tmx>'
     )
+
+
+def get_m2m_changes(current_qs, new_qs):
+    """
+    Get difference between states of m2m relation.
+    When formfield_items is empty, return all items from object_items as removed.
+
+    :arg django.db.models.QuerySet `current_qs`: objects from the current state of relation.
+    :arg django.db.models.QuerySet `final_qs`: objects from the future state of m2m
+    :returns: A tuple with 2 querysets for added and removed items from m2m
+    """
+
+    add_items = new_qs.exclude(
+        pk__in=current_qs.values_list('pk', flat=True)
+    )
+
+    remove_items = (
+        current_qs.exclude(
+            pk__in=new_qs.values_list('pk', flat=True)
+        ).exclude(
+            pk__in=add_items.values_list('pk', flat=True)
+        )
```

Comment:
Why do you exclude things from ``add_items``? I don't understand what that adds here. `current - new` should be enough.

Reply:
I actually removed that locally, it's an artifact from different design.